src/kqueue.c

```diff
@@ -180,8 +180,17 @@ static int kq_set_deadline(void* state, uint64_t deadline)
 	struct kq_state* self = state;
 
 	struct kevent event;
+#ifdef __MACH__
+	struct timespec ts = { 0 };
```

Comment:
I am not really proud about this workaround. As far as I can see, is that on each call, the current time was also measured somewhere and we could introduce a new parameter. But I wouldn't want to do such a breaking change without asking before.